### PR TITLE
Enable Windows stack protection (/GS flag)

### PR DESCRIPTION
**Pull request by theresa-m**
Performance summary:
- Throughput: no regression
- Compilation: 2% more CPU spent
- Startup: no regression
- Footprint: no regression

Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

File: runtime/codert_vm/cnathelp.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ buildJITResolveFrameWithPC(J9VMThread *currentThread, UDATA flags, UDATA parmCou
 	currentThread->jitException = NULL;
 	resolveFrame->specialFrameFlags = flags;
 #if defined(J9SW_JIT_HELPERS_PASS_PARAMETERS_ON_STACK)
-	resolveFrame->parmCount = parmCount;
+	resolveFrame->parmCount = 0;
```
Comment (fjeremic) [inline runtime/codert_vm/cnathelp.cpp]: Is this change intentional? If so then the `#if` and `#else` paths are identical and can be merged.

Comment (DanHeidinga) [inline runtime/codert_vm/cnathelp.cpp]: Why this change?  Is it required by the /GS flag?  

Changing this indicates that the resolveFrame doesn't have any parameters, which means we won't attempt to walk them / update them / etc.

`J9SW_JIT_HELPERS_PASS_PARAMETERS_ON_STACK` is like a x86-32 thing.  Does /GS apply to both 32/64?

**Comment by DanHeidinga** [comment]
@theresa-m you did a bunch of startup / throughput / footprint analysis on this change.  Can you summarize that here?

**Comment by andrewcraik** [comment]
Does this change affect all components of J9 or just the VM/GC? If it includes the JIT measurements of the compile-time impact (amount of time it takes the JIT to compile code) would also be good to see.

File: runtime/codert_vm/cnathelp.cpp (hunk 1 above)
Comment (theresa-m) [reply runtime/codert_vm/cnathelp.cpp]: The change was attempting to address this concern:
>Some of the JIT helpers (c code) run on the java stack now in c interpreter.  We'd need to keep the flag disabled for those helpers.  The file to disable the option for is cnathelp.cpp

/GS applies to all platforms according to its documentation.

**Comment by theresa-m** [comment]
Performance summary (so far):
- Throughput: no regression
- Compilation: 2% more CPU spent
- Startup: no regression
- Footprint: no regression

@andrewcraik I assume that compile time number is what you are referring to? @mpirvu was okay with this number.

File: runtime/codert_vm/cnathelp.cpp (hunk 1 above)
Comment (DanHeidinga) [reply runtime/codert_vm/cnathelp.cpp]: That should result in makefile changes to disable the /GS flag for those files.

**Comment by andrewcraik** [comment]
@theresa-m yes FYI @vijaysun-omr due to the CPU increase

File: runtime/codert_vm/cnathelp.cpp (hunk 1 above)
Comment (theresa-m) [reply runtime/codert_vm/cnathelp.cpp]: ah.. that makes more sense. thanks!

**Comment by vijaysun-omr** [comment]
Yes I am okay with the CPU increase reported.

File: runtime/codert_vm/cnathelp.cpp (hunk 1 above)
Comment (theresa-m) [reply runtime/codert_vm/cnathelp.cpp]: Made changes to correct this in the last push.

**Comment by charliegracie** [comment]
I see a similar PR agains the OMR project.  Do these changes have to be co-ordinated in anyway?

**Comment by theresa-m** [comment]
@charliegracie it won't break anything to have one in and not the other

**Comment by pshipton** [comment]
@DanHeidinga any further concerns?

**Comment by DanHeidinga** [comment]
Merging without a build as we don't have PR builds on windows yet.
